<!-- -->
<launch>
  <!-- send table.xml to param server -->
  <arg name="car" default="true" />
  <arg name="pedestrian" default="false" />
  <arg name="obj_car" default="obj_car" />
  <arg name="obj_person" default="obj_person" />


  <group ns="sync_obj_fusion">
    <group if="$(arg car)">
      <group ns="obj_car">
        <node pkg="synchronization" type="sync_obj_fusion" name="sync_$(arg obj_car)_fusion">
          <remap from="/obj_label" to="/$(arg obj_car)/obj_label"/>
          <remap from="/cluster_centroids" to="/cluster_centroids"/>
          <remap from="/obj_pose" to="/$(arg obj_car)/obj_pose"/>
          <remap from="/sync_obj_fusion/obj_label" to="/sync_obj_fusion/$(arg obj_car)/obj_label"/>
          <remap from="/sync_obj_fusion/cluster_centroids" to="/sync_obj_fusion/$(arg obj_car)/cluster_centroids"/>
        </node>
      </group>
    </group>

    <group if="$(arg pedestrian)">
      <group ns="obj_person">
        <node pkg="synchronization" type="sync_obj_fusion" name="sync_$(arg obj_person)_fusion">
          <remap from="/obj_label" to="/$(arg obj_person)/obj_label"/>
          <remap from="/cluster_centroids" to="/cluster_centroids"/>
          <remap from="/obj_pose" to="/$(arg obj_person)/obj_pose"/>
          <remap from="/sync_obj_fusion/obj_label" to="/sync_obj_fusion/$(arg obj_person)/obj_label"/>
          <remap from="/sync_obj_fusion/cluster_centroids" to="/sync_obj_fusion/$(arg obj_person)/cluster_centroids"/>
        </node>
      </group>
    </group>
  </group>
</launch>
